tools/cli/workflowCommands.go

```diff
@@ -298,74 +297,69 @@ func formatInputsForDisplay(inputs []interface{}) string {
+
+	var saValues []string
+	if c.IsSet(FlagSearchAttributesValue) {
+		saValues = c.StringSlice(FlagSearchAttributesValue)
+	} else if c.IsSet(FlagSearchAttributesKey) {
+		ErrorAndExit(fmt.Sprintf("Search attribute values must be provided using %s.", FlagSearchAttributesValue), nil)
+	} else {
 		return nil
 	}
 
-	if len(searchAttrKeys) != len(rawSearchAttrVals) {
-		ErrorAndExit(fmt.Sprintf("Uneven number of search attributes keys (%d): %v and values(%d): %v.", len(searchAttrKeys), searchAttrKeys, len(rawSearchAttrVals), rawSearchAttrVals), nil)
+	if len(saKeys) != len(saValues) {
+		ErrorAndExit(fmt.Sprintf("Number of search attributes keys %d and values %d are not equal.", len(saKeys), len(saValues)), nil)
 	}
 
-	fields := make(map[string]interface{}, len(searchAttrKeys))
+	fields := make(map[string]interface{}, len(saKeys))
 
-	for i, v := range rawSearchAttrVals {
+	for i, saValue := range saValues {
 		var j interface{}
-		if err := json.Unmarshal([]byte(v), &j); err != nil {
+		if err := json.Unmarshal([]byte(saValue), &j); err != nil {
 			ErrorAndExit("Search attribute JSON parse error.", err)
 		}
-		fields[searchAttrKeys[i]] = j
+		fields[saKeys[i]] = j
 	}
 
 	return fields
 }
 
 func unmarshalMemoFromCLI(c *cli.Context) map[string]interface{} {
-	rawMemoKey := c.String(FlagMemoKey)
-	var memoKeys []string
-	if strings.TrimSpace(rawMemoKey) != "" {
-		memoKeys = strings.Split(rawMemoKey, " ")
+	if !c.IsSet(FlagMemoKey) && (c.IsSet(FlagMemo) || c.IsSet(FlagMemoFile)) {
+		ErrorAndExit(fmt.Sprintf("Memo keys must be provided using %s.", FlagMemoKey), nil)
 	}
 
-	jsonsRaw := readJSONInputs(c, jsonTypeMemo)
-	if len(jsonsRaw) == 0 {
-		return nil
-	}
-	rawMemoValue := string(jsonsRaw[0]) // StringFlag may contain up to one json
+	memoKeys := c.StringSlice(FlagMemoKey)
 
-	if err := validateJSONs(rawMemoValue); err != nil {
-		ErrorAndExit("Input is not valid JSON, or JSONs concatenated with spaces/newlines.", err)
+	if c.IsSet(FlagMemo) && c.IsSet(FlagMemoFile) {
+		ErrorAndExit(fmt.Sprintf("Only one of %s or %s should be used.", FlagMemo, FlagMemoFile), nil)
 	}
 
-	var memoValues []interface{}
-
-	var sc fastjson.Scanner
-	sc.Init(rawMemoValue)
-	for sc.Next() {
-		memoValues = append(memoValues, sc.Value())
-	}
-	if err := sc.Error(); err != nil {
-		ErrorAndExit("Memo JSON parse error.", err)
+	var memoValues []string
+	if c.IsSet(FlagMemoFile) {
+		inputFile := c.String(FlagMemoFile)
+		// This method is purely used to parse input from the CLI. The input comes from a trusted user
+		// #nosec
+		data, err := os.ReadFile(inputFile)
+		if err != nil {
+			ErrorAndExit(fmt.Sprintf("Error reading memo file %s.", inputFile), err)
+		}
+		memoValues = strings.Split(string(data), "\n")
+	} else if c.IsSet(FlagMemo) {
+		memoValues = c.StringSlice(FlagMemo)
+	} else if c.IsSet(FlagMemoKey) {
+		ErrorAndExit(fmt.Sprintf("Memo values must be provided using %s or %s.", FlagMemo, FlagMemoFile), nil)
+	} else {
+		return nil
```

Comment:
Same comment apply here, should do all these check at the beginning of this func and make it more easy to read.
```
isKeySet := ...
isMemoSet := ...
isFileSet := ...
isValueSet := isMemoSet || isFileSet...
if !isKeySet && !isValueSet {
  return nil
} else if isKeySet && !isValueSet {
  ErrorAndExit(...)
} else if !isKeySet && isValueSet {
  ErrorAndExit(...)
} else if isMemoSet && isFileSet {
  ErrorAndExit(...)
} 
memoKeys := c.StringSlice(FlagMemoKey)
memoValues := c.StringSlice(FlagMemo)
if isFileSet {
   memoValues = ...
}
...
```